File: 2017.js/day1.js
export function Day1(input){
    const start = performance.now();
    console.log(`    1: ${Part1(input).toString().padStart(15)} / ${Part2(input).toString().padEnd(15)}  ${performance.now()-start}`);
}

function Part1(input){
    var sum = 0
    for(var i = 0; i < input.length; i++){
        if(input[i] == input[(i+1)%input.length]) sum += parseInt(input[i]);
    }
    return sum;
}

function Part2(input){
    var sum = 0
    for(var i = 0; i < input.length; i++){
        if(input[i] == input[(i+input.length/2)%input.length]) sum += parseInt(input[i]);
    }
    return sum;
}

File: 2021.kt/day1.kt
package aoc

import kotlin.time.measureTime
import kotlin.collections.*

@kotlin.time.ExperimentalTime
fun day1(input: List<String>){
    val exec = measureTime { 
        val nums: List<Int> = input.map{it.toInt()}.toList();
        print("    1: ${part1(nums).toString().padStart(15)} / ${part2(nums).toString().padEnd(15)}  ")
    }
    println(exec.inWholeMilliseconds);
}

private fun part1(input: List<Int>): Int{
    var sum = 0;
    for(i in 1..input.size-1){
        if(input[i]>input[i-1])
            sum++
    }
    return sum;
}

private fun part2(input: List<Int>): Int{
    var sum = 0;
    for(i in 3..input.size-1){
        if((input[i]+input[i-1]+input[i-2])>(input[i-1]+input[i-2]+input[i-3]))
            sum++
    }
    return sum;
}